ibis/backends/sql/compilers/datafusion.py

```diff
@@ -136,6 +135,8 @@ def visit_Cast(self, op, *, arg, to):
         if to.is_timestamp():
             return self._to_timestamp(arg, to)
         if to.is_decimal():
+            from ibis.formats.pyarrow import PyArrowType
+
             return self.f.arrow_cast(arg, f"{PyArrowType.from_ibis(to)}".capitalize())
```

Comment:
This isn't 100% free of dependencies here, because we're using the pyarrow type system for casts, because datafusion doesn't have any other way to represent types than their pyarrow string representation. So, unfortunately, if you're converting from non-datafusion dialects to datafusion then you'll need pyarrow installed.

Reply:
I think this is a big improvement and this will only trip up users if they are going from non-datafusion dialects to datafusion AND if they have decimal columns.  :shipit:

Reply:
Yep, it's a somewhat specific use case I think.